### ISSUE TITLE
missing query function

### ISSUE THREAD

**Issue by sebglon** [AGBrown/pouchdb.d.ts#33]
I have not found 'query function.
How can i write query map/reduce?

**Comment by AGBrown** [comment]
@sebglon Query hasn't been added to the d.ts yet. If you would like to add it then I'll gladly accept a PR.

**Comment by AGBrown** [comment]
I would like to see if @filionf can add his findings to this issue. From the network it looks like he has added the query function with [c63f5f6](https://github.com/AGBrown/pouchdb.d.ts/commit/c63f5f6#diff-20ca7868)

**Comment by filionf** [comment]
Yes, I added query to my repo for my own purpose, but I believe I had some elements missing on it, which is why I have not sent a PR. I could have another look, which branch should I be targeting for this?

**Comment by AGBrown** [comment]
@filionf You'll be fine if you work off the current `dev` at e0147ee. I _think_ that the current `i23-ctornames` will actually get merged into dev, but it's a bit of a wip at the moment (still).
